cli/worker.rs

```diff
@@ -344,12 +347,39 @@ impl CliMainWorkerFactory {
     let (main_module, is_main_cjs) = if let Ok(package_ref) =
       NpmPackageReqReference::from_specifier(&main_module)
     {
+      // For main_module that exists in package.json deps,
+      // use the version defined in package.json instead of adding new dependency.
+      let existing_package_req = shared
+        .options
+        .maybe_package_json_deps
+        .as_ref()
+        .and_then(|deps| {
+          let PackageReq { version_req, name } = package_ref.req();
+          deps
+            .get(name)
+            .filter(|existing_req| {
+              version_req.version_text() == "*" && existing_req.is_ok()
+            })
+            .map(|existing_req| {
+              let existing_req = existing_req.as_ref().unwrap();
+              existing_req
+            })
+        });
       shared
         .npm_resolver
-        .add_package_reqs(&[package_ref.req().clone()])
+        .add_package_reqs(&[existing_package_req
+          .unwrap_or_else(|| package_ref.req())
+          .clone()])
         .await?;
-      let node_resolution =
-        self.resolve_binary_entrypoint(&package_ref, &permissions)?;
+      let node_resolution = self.resolve_binary_entrypoint(
+        &(existing_package_req
+          .map(|req| {
+            NpmPackageReqReference::from_str(format!("npm:{}", req).as_ref())
```

Comment:
I believe this should also include the specified sub path? (`package_ref.sub_path()` or something like that)

Reply:
Thanks for your review,  updated now ~